invokeai/backend/model_manager/probe.py

```diff
@@ -321,10 +324,19 @@ def _get_checkpoint_config_path(
             return possible_conf.absolute()
 
         if model_type is ModelType.Main:
-            config_file = LEGACY_CONFIGS[base_type][variant_type]
-            if isinstance(config_file, dict):  # need another tier for sd-2.x models
-                config_file = config_file[prediction_type]
-            config_file = f"stable-diffusion/{config_file}"
+            if base_type == BaseModelType.Flux:
+                # TODO: Decide between dev/schnell
+                checkpoint = ModelProbe._scan_and_load_checkpoint(model_path)
+                state_dict = checkpoint.get("state_dict") or checkpoint
+                if "guidance_in.out_layer.weight" in state_dict:
+                    config_file = "flux-dev"
+                else:
+                    config_file = "flux-schnell"
```

Comment:
It feels like we are mis-using the config_path field here. Shouldn't dev vs schnell just be a new property that's specific to FLUX models?

Reply:
Yeah, I mostly set these since they're a required field on the CheckpointConfigBase. In my head I justified it since they're a key in the flux param config. A kind of a "config path", but perhaps it's stretching it too far since it's generally considered a filepath elsewhere in the codebase.

Reply:
Ya, maybe we should just subclass directly from ModelConfigBase? I don't think we'd want anyone to actually use the config_path field for FLUX models, so best not to set it.

Reply:
We discussed this in a call separate to this PR. Punting separating the configs to a future date when we have a better answer to this. A TODO comment has been added to reflect this decision